frontend/src/components/app/ErrorScreen.js

```diff
@@ -1,34 +1,67 @@
 import counterpart from 'counterpart';
 import React, { Component } from 'react';
-
-const OFFLINE_MESSAGE_LINE1 = 'Connection lost.';
-const OFFLINE_MESSAGE_LINE2 =
-  'There are some connection issues. ' +
-  'Check connection and try to refresh the page.';
-
+import PropTypes from 'prop-types';
+import { checkLoginRequest } from '../../api/login';
+import { CONNECTION_ERROR_RETRY_INTERVAL_MILLIS } from '../../constants/Constants';
+import { getCurrentActiveLanguage } from '../../utils/locale';
 class ErrorScreen extends Component {
   constructor(props) {
     super(props);
+    this.intervalId = null;
+    this.activeLang = getCurrentActiveLanguage();
   }
-  render() {
-    let line1 = counterpart.translate('window.error.noStatus.title');
-    let line2 = counterpart.translate('window.error.noStatus.description');
 
-    if (!line1) {
-      line1 = OFFLINE_MESSAGE_LINE1;
-    }
+  pingServer = async () => {
+    checkLoginRequest()
+      .then((response) => {
+        if (response && response.status === 200) {
+          window.location.reload();
+        }
+      })
+      .catch((e) => console.error('checkLoginRequest error:', e));
+  };
 
-    if (!line2) {
-      line2 = OFFLINE_MESSAGE_LINE2;
+  componentDidMount() {
+    const { errorType } = this.props;
+    if (errorType) {
+      this.intervalId = setInterval(
+        this.pingServer,
+        CONNECTION_ERROR_RETRY_INTERVAL_MILLIS
+      );
     }
+  }
+
+  componentWillUnmount() {
+    this.intervalId && clearInterval(this.intervalId);
+  }
+
+  render() {
+    const { errorType } = this.props;
+    const title = counterpart.translate(`window.error.${errorType}.title`, {
+      fallback: counterpart.translate(
```

Comment:
why this is needed?

i thought that we are plugining in multiple sources to counterpart
and counterpart will do the magic.

At this level i would not expect to bother about from where the message comes.

If we have to do like this for any other case, our whole code will be poluted.

Reply:
I don't understand. the counterpart is doing the magic. If there is no key window.error.${errorType}.title we are going to use the offline existing key using the counterpart fallback mechanism

<img width="1296" alt="Screenshot 2021-12-09 at 12 18 24" src="https://user-images.githubusercontent.com/1708561/145378133-d898fdd3-5cdd-42bb-879d-823e3d74ac35.png">

Reply:
![image](https://user-images.githubusercontent.com/1244701/145402592-a82b9edc-111d-4322-bb16-eb61fe64a67e.png)